src/main/java/com/artipie/rpm/Rpm.java

```diff
@@ -185,30 +191,31 @@ public Completable batchUpdate(final Key prefix) {
                         Files.createTempFile("repomd-", ".xml")
                     );
                     repomd.begin(System.currentTimeMillis() / Tv.THOUSAND);
-                    return new Repository(
-                        repomd,
-                        Arrays.asList(
-                            new MetadataFile(
-                                "primary",
-                                new PrimaryOutput(Files.createTempFile("primary-", ".xml"))
-                                    .start(),
-                                repomd
-                            ),
-                            new MetadataFile(
-                                "others",
-                                new OthersOutput(Files.createTempFile("others-", ".xml"))
-                                    .start(),
-                                repomd
-                            ),
+                    final List<MetadataFile> files = Arrays.asList(
```

Comment:
@carlosmiranda let's extract the whole `reduceWith` body to a private method to make `batchUpdate` more readable.